lib/1.4/dml-builtins.dml

```diff
@@ -539,6 +539,20 @@ template device {
     param objtype = "device";
 
     param simics_api_version auto;
+    param _api_5_or_older = simics_api_version == "5"
+        || simics_api_version == "4.8";
+    param _api_6_or_older = simics_api_version == "6" || _api_5_or_older;
+
+    param _DEPRECATE_PORT_PROXY_IFACES auto;
+    #if (!_DEPRECATE_PORT_PROXY_IFACES && !_api_6_or_older) {
+        error "The PORT_PROXY_IFACES feature can only be enabled"
+            + " with API=6 or older";
+    }
+    param _DEPRECATE_PORT_PROXY_ATTRS auto;
```

Comment:
The naming scheme can be debated -- we don't use UPPER_SNAKE_CASE for other params, the uppercase is inherited from the command-line tags, which are uppercase for consistency with warning and error tags. I'm not sure that consistency is important; we could just lowercase deprecation tags. We could also mix-case, as in `_deprecate_PORT_PROXY_ATTRS`.  Or just keep it uppercase; there is uncontroversial widespread use of uppercase param names, in particular for simulation-related constants or array size parameters.

Reply:
My vote is mixed case, but I agree that whatever choice we pick is likely to be uncontroversial.